### PR TITLE
issue #3341:fix TextureCache::addImageAsync repeatedly execute for the same image

### PR DESCRIPTION
**Pull request by WenhaiLin**

### CODE REVIEW

File: cocos/2d/CCTextureCache.cpp [hunk 1]
```diff
@@ -170,17 +170,40 @@ void TextureCache::loadImage()
...
+        auto it = _textures.find(asyncStruct->filename);
+        if( it == _textures.end() )
         {
-            CC_SAFE_RELEASE(image);
-            CCLOG("can not load %s", filename);
-            continue;
+           _imageInfoMutex.lock();
+           ImageInfo *imageInfo;
+           unsigned int pos = 0;
+           unsigned int infoSize = _imageInfoQueue->size();
```
Comment (minggo) [inline cocos/2d/CCTextureCache.cpp]: Is `size()` return `unsigned int`?

Comment (minggo) [inline cocos/2d/CCTextureCache.cpp]: I think it should be `size_t` here.

Comment (WenhaiLin) [inline cocos/2d/CCTextureCache.cpp]: std：size_t?

Comment (minggo) [inline cocos/2d/CCTextureCache.cpp]: size_t or std::size_t both are ok.

**Pull request merged**